pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -326,11 +326,31 @@ func (c *Cluster) postReconcileUpdateOSDProperties(desiredOSDs map[int]*OSDInfo)
 	}
 	logger.Debugf("post processing osd properties with %d actual osds from ceph osd df and %d existing osds found during reconcile", len(osdUsage.OSDNodes), len(desiredOSDs))
 	for _, actualOSD := range osdUsage.OSDNodes {
-		if desiredOSD, ok := desiredOSDs[actualOSD.ID]; ok {
-			if err := c.updateDeviceClassIfChanged(actualOSD.ID, desiredOSD.DeviceClass, actualOSD.DeviceClass); err != nil {
-				// Log the error and allow other updates to continue
-				logger.Error(err)
-			}
+		desiredOSD, ok := desiredOSDs[actualOSD.ID]
+		if !ok {
+			return nil
+		}
+		if err := c.resizeOsdCrushWeight(actualOSD); err != nil {
+			// Log the error and allow other updates to continue
+			logger.Error(err)
+		}
+		if err := c.updateDeviceClassIfChanged(actualOSD.ID, desiredOSD.DeviceClass, actualOSD.DeviceClass); err != nil {
+			// Log the error and allow other updates to continue
+			logger.Error(err)
```

Comment:
Because these errors are only logged, let's also be sure to include the namespace of the Ceph cluster that corresponds to the error by appending that. Otherwise, we might wind up with scenarios where we can't tell which cluster is giving the error.

```go
logger.Error(fmt.Sprintf("failed to XYZ on cluster in namespace %q: %s", namespace, err))
```